packages/core/src/agent/index.ts

```diff
@@ -1119,6 +1120,47 @@ export class Agent<
     };
   }
 
+  /**
+   * Adds response messages from a step to the MessageList and schedules persistence.
+   * This is used for incremental saving: after each agent step, messages are added to a save queue
+   * and a debounced save operation is triggered to avoid redundant writes.
+   *
+   * @param result - The step result containing response messages.
+   * @param messageList - The MessageList instance for the current thread.
+   * @param threadId - The thread ID.
+   * @param memoryConfig - The memory configuration for saving.
+   * @param runId - (Optional) The run ID for logging.
+   */
+  private async saveStepMessages({
+    result,
+    messageList,
+    threadId,
+    memoryConfig,
+    runId,
+  }: {
+    result: any;
+    messageList: MessageList;
+    threadId?: string;
+    memoryConfig?: MemoryConfig;
+    runId?: string;
+  }) {
+    try {
+      messageList.add(result.response.messages, 'response');
+      const memory = this.getMemory();
+      const saveQueueManager = new SaveQueueManager({
```

Comment:
shouldn't we create the SaveQueueManager instance in __primitive and pass it around? Looks like here it'll create a new instance after each step unless I'm misreading

Reply:
oh there is one, maybe we just need to pass it into this method then